### PR TITLE
feat(warden): add PQR checks

### PR DESCRIPTION
**Pull request by Helias**
ported some checks from: https://github.com/TrinityCore/TrinityCore/commit/0531463a3c02d98ed77e9c7689ad7237ecaddc09#

## CHANGES PROPOSED:
-  alter the table warden_checks, modify the column `str` from `varchar(20)` to `varchar(170)`
- add some warden checks

## ISSUES ADDRESSED:
- related to https://github.com/azerothcore/azerothcore-wotlk/issues/2004

## TESTS PERFORMED:
- tested in-game with and without these checks

## HOW TO TEST THE CHANGES:
- open WoW
- run [these PQR scripts](https://mega.nz/file/s4AmyICA#jw_DDeNbVaS6hB0nJd_3gWhz8P14KQHfGzUuGMcZ37c)
- the server will ban you

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/scripts/Northrend/IcecrownCitadel/boss_deathbringer_saurfang_TC.cpp [hunk 1]
```diff
@@ -0,0 +1,1445 @@
+// /*
```
Comment (Nefertumm) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_deathbringer_saurfang_TC.cpp:1]: mistake?

**Comment by Nefertumm** [comment]
This is awesome!

**Comment by CosmeF** [comment]
> 
> 
> This is awesome!

Totally! Thanks a lot @Helias 

I'll be performing some tests, and keep you posted

File: src/server/scripts/Northrend/IcecrownCitadel/boss_deathbringer_saurfang_TC.cpp (hunk 1 above)
Comment (Helias) [reply src/server/scripts/Northrend/IcecrownCitadel/boss_deathbringer_saurfang_TC.cpp:1]: yes, absolutely

Comment (Helias) [reply src/server/scripts/Northrend/IcecrownCitadel/boss_deathbringer_saurfang_TC.cpp]: removed

**Comment by Helias** [comment]
The PQR checks are really necessary, but it's "impossible" to check the PQR with the 100% accuracy of detection because it is mainly a client-side script.
Now, I am not sure that this "works", I am considering to import the entire Warden from TC, that should be better, but I am not sure if it is fesible.

About this PR, the only thing that I miss to check is this
```C++
    // std::from_chars(msg.data() + sizeof(WARDEN_TOKEN) - 1, msg.data() + msg.size(), id, 10);
```
if I migrate even this string to AC and it is not working properly, I think that we can't migrate only the PQR LUA check but we need to migrate all the warden (not sure ofc, but it's a guess).

File: src/server/game/Warden/Warden.cpp [hunk 2]
```diff
@@ -238,6 +240,78 @@ std::string Warden::Penalty(WardenCheck* check /*= NULL*/, uint16 checkFailed /*
...
+        }
+        case WARDEN_ACTION_LOG:
+        default:
+            return/*  "None" */;
+    }
+
+    // return EnumUtils::ToTitle(action);
+}
+
+
```
Comment (Winfidonarleyan) [inline src/server/game/Warden/Warden.cpp:314]: delete this line

Comment (Helias) [reply src/server/game/Warden/Warden.cpp]: done

**Comment by Helias** [comment]
@CosmeF if you can give me some feedbacks, because I am not sure if we will merge this to master.
I am not sure if this PR improve the PQR checks or not, testing locally the PQR they still work :(

**Comment by Viste** [comment]
@Helias if you want get it work maybe you need request something from client module?
https://github.com/azerothcore/azerothcore-wotlk/blob/b397f4292940d181fb11c0f84f7c2ad2449b73b9/src/server/game/Warden/WardenWin.cpp#L299
https://github.com/TrinityCore/TrinityCore/commit/0531463a3c02d98ed77e9c7689ad7237ecaddc09#diff-ed73961ed1e9b5b0c2aee482b116a916R223
;))

**Comment by BarbzYHOOL** [comment]
@CosmeF ??? so?

**Comment by Helias** [comment]
@sanctum32 Is improving or rewriting this, this is a WIP PR, don't need test for now

**Comment by Helias** [comment]
closing this PR for https://github.com/azerothcore/azerothcore-wotlk/pull/3875
@sanctum32 did an awesome job about the PQR checks.

**Pull request closed without merging**